File: src/app/types/employee.ts


/*export interface IEmployee {
  id: number;
  name: string;
  email: string;
  phone: string;
  jobTitle: string;
  gender: string;
  departmentId: number;
  joiningDate: string;        // Format: YYYY-MM-DDTHH:mm:ss (ISO 8601)
  lastWorkingDate: string;
  dateOfBirth: string;
}

*/

// ✅ src/app/types/employee.ts

/*export interface IEmployee {
  id: number,
  name: string;
  email: string;
  phone: string;
  jobTitle: string;
  gender: string;
  departmentId: number;
  joiningDate: string;
  lastWorkingDate: string;
  dateOfBirth: string;
}

*/
export interface ICreateEmployee {
  name: string;
  email: string;
  phone?: string;
  jobTitle: string;
  gender: string;
  departmentId: number;
  joiningDate: string;      // Use string to match ISO format from date picker
  lastWorkingDate: string;
  dateOfBirth: string;
}

export interface IEmployee extends ICreateEmployee {
  id: number;
}

File: Entity/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeeManagement22.Entity
{
    public class Employee
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [EmailAddress]
        public string Email { get; set; }

        public string Phone { get; set; }
        public string JobTitle { get; set; }
        public string Gender { get; set; }

        // Foreign key to Department
        public int? DepartmentId { get; set; }

        // ✅ Correct navigation property
        [ForeignKey("DepartmentId")]
        public Department? Department { get; set; }

        public DateTime JoiningDate { get; set; }
        public DateTime LastWorkingDate { get; set; }
        public DateTime DateOfBirth { get; set; }

        public int? UserId { get; set; }
        public User? User { get; set; }
    }

}
